lib/ci-helper.ts

```diff
@@ -763,6 +772,211 @@ export class CIHelper {
+                        node?.commit.author?.user?.login;
+
+                    // login may not be present if email unknown to github
+                    if (login && pr.author !== login) {
+                        throw new Error("Request author is not commit author.");
+                    }
+                });
+            }
+
+            const files = prInfo.files;
+            if (files && files.totalCount && files.nodes) {
+                files.nodes.map(node => {
+                    const changedFile = node as PullRequestChangedFile;
+                    this.checkNewUserFile(changedFile, files.totalCount,
+                        genLabel);
+                });
+            } else {
+                throw new Error("No files updated.");
+            }
+
+            const queryRef = `query {
+                repository(name: "${pr.baseRepo}", owner: "${pr.author}") {
+                    ref(qualifiedName: "${prInfo.baseRefName}") {
+                        name
+                    }
+                }
+            }`;
+
+            const refInfo = await this.github.graphql<queryResponse>(
+                pr.baseOwner,
+                queryRef
+            );
+
+            if (!refInfo || !refInfo.repository.ref) {
+                throw new Error(`Base branch ${prInfo.baseRefName
+                    } not found in ${pr.author}/${pr.baseRepo}.`);
+            }
+        } catch (e) {
+            if ((e as labelInfo).label === undefined ) {
+                const error = e as Error;
+                result = false;
+                await this.github.closePR(prInfo.url, `PR is being closed:\n\n${
+                    error.message}`);
+            } else {
+                const label = e as labelInfo;
+                await this.github.addPRLabels(prInfo.url, [label.label]);
+                await this.github.addPRComment(prInfo.url, `Label \`${
+                    label.label}\` added for this reason:\n\n${
+                    label.message}`);
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Perform some validation tests on a file in a pull request.
+     *
+     * @param changedFile file that is part of a commit in a pull request
+     * @param files number of files that are changed in the pull request
+     */
+    public checkNewUserFile(changedFile: PullRequestChangedFile,
+        files: number, setLabel: setLabelInfo):
+        void {
+        const filenameParts = path.parse(changedFile.path);
+
+        // one file with only deletions?
+        if (files === 1 &&
+            !changedFile.additions && changedFile.deletions) {
+            throw setLabel("validate", `File \`${changedFile.path
+                }\` expecting more than one change.`)
+        }
+
+        if (filenameParts.name.match(/.+\./)) {
+            throw new Error(`File \`${changedFile.path
+                }\` not allowed with multi-part name`);
+        }
+
+        if (filenameParts.name.match(/ /)) {
```

Comment:
I think we'll need to be more precise here:

```shell-session
$ git ls-files *\ *
t/t4135/add-with backslash.diff
t/t4135/add-with quote.diff
t/t4135/add-with spaces.diff
t/t4135/add-with tab.diff
t/t4135/diff-with backslash.diff
t/t4135/diff-with quote.diff
t/t4135/diff-with spaces.diff
t/t4135/diff-with tab.diff
t/t4135/git-with backslash.diff
t/t4135/git-with quote.diff
t/t4135/git-with spaces.diff
t/t4135/git-with tab.diff
```

So maybe something like `/^(?!t\/t[0-9]{4}\/).* /`, i.e. with a negative look-ahead that would force any path starting with `t/tNNNN/` to be excluded from the check?

Reply:
Okay.

Reply:
>  Adding check for `dir` starting with `t/`.

More like: Adding a check that the path _does not_ start with `t/t\d\d\d\d/` ;-) (because in the support directories of the tests, the space is legitimate in file names).